### PR TITLE
feature: TRACEFOSS-604 Dashboard extension

### PR DESCRIPTION
**Pull request by ds-ext-sceronik**

### CODE REVIEW

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/common/support/AlertsSupport.java [hunk 1]
```diff
@@ -53,16 +54,27 @@ public Long defaultReceivedAlertStored() {
         return storedAlert(entity);
     }
 
-    Long defaultAcknowledgedAlertStored() {
+    public Long defaultReceivedAlertStoredWithAssets(List<AssetAsBuiltEntity> assets) {
```
Comment (mkanal) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/common/support/AlertsSupport.java:57]: this method name is not clear to me? 
receivedAlertsForAssets ?

**Review by mkanal — CHANGES_REQUESTED** [review]

File: tx-models/src/main/java/assets/response/DashboardResponse.java [hunk 2]
```diff
@@ -23,10 +23,18 @@
 
 public record DashboardResponse(
         @ApiModelProperty(example = "5")
-        Long myItems,
+        Long myParts,
```
[the remark this answers is not in the document]
Comment (ds-ext-sceronik) [reply tx-models/src/main/java/assets/response/DashboardResponse.java:26]: this constructor is default for record in java, providing custom constructor have its limitations and requires calling super constructor if used which will just make it less clear on what is happening

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/common/support/AlertsSupport.java [hunk 3]
```diff
@@ -53,16 +54,27 @@ public Long defaultReceivedAlertStored() {
...
+        return defaultReceivedAlertStoredWithAssets(QualityNotificationStatusBaseEntity.RECEIVED, assets);
+    }
+
+    public Long defaultSentAlertStoredWithAssets(List<AssetAsBuiltEntity> assets) {
```
[the remark this answers is not in the document]
Comment (ds-ext-sceronik) [reply tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/common/support/AlertsSupport.java:61]: support class for alerts provides predefined alerts considered as default for integration testing. all support classes providing this have default prefix

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/common/support/AlertsSupport.java (hunk 1 above)
Comment (ds-ext-sceronik) [reply tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/common/support/AlertsSupport.java:57]: support method providing default alert for Integration test purpouse . we use default for all predefined data for support classes. 
I fill change name with For as it seems better as You suggested :) 
defaultReceivedAlertStoredForAssets()

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/assets/DashboardControllerIT.java [hunk 4]
```diff
@@ -74,9 +84,43 @@ void givenRoles_whenGetDashboard_thenReturnResponse(final List<JwtRole> roles) t
                 .log().all()
                 .when().get("/api/dashboard")
                 .then().statusCode(200)
-                .body("myItems", equalTo(1))
+                .body("myParts", equalTo(1))
+                .body("otherParts", equalTo(12))
```
[the remark this answers is not in the document]
Comment (ds-ext-sceronik) [reply tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/assets/DashboardControllerIT.java:88]: For controller I think it is enough to see database flow. 
for specific count for assets there are these edgecases tests implemented in AlertsRepositoryIT.java

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/qualitynotification/infrastructure/alert/repository/AlertsRepositoryImpl.java [hunk 5]
```diff
@@ -120,6 +123,18 @@ public long countQualityNotificationEntitiesByStatus(QualityNotificationStatus q
         return jpaAlertRepository.countAllByStatusEquals(QualityNotificationStatusBaseEntity.valueOf(qualityNotificationStatus.name()));
     }
 
+    @Transactional
+    @Override
+    public long countDistinctAffectedPartsWhereStatusInAndOwnerEqual(List<QualityNotificationStatus> statuses, Owner owner) {
+        return jpaAlertRepository.findAllByStatusIn(QualityNotificationStatusBaseEntity.from(statuses))
```
[the remark this answers is not in the document]
Comment (ds-ext-sceronik) [reply tx-backend/src/main/java/org/eclipse/tractusx/traceability/qualitynotification/infrastructure/alert/repository/AlertsRepositoryImpl.java:129]: this is actual count logic using streams. mapping is done only for sole purpouse of counting . 
building custom querry did not work as expected when counting so had to do count logic within this method. 

this method is tested with edge cases 2 alerts with the same asset references. 
for debugging it is actually much easier to do when java stream is used

[the remark this answers is not in the document]
Comment (ds-ext-sceronik) [reply tx-backend/src/main/java/org/eclipse/tractusx/traceability/qualitynotification/infrastructure/alert/repository/AlertsRepositoryImpl.java:129]: for suggestion I cannot change it this is Spring data JPA generated query with methodName so specific format need to be maintained for it to work

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/assets/application/rest/DashboardControllerTest.java [hunk 6]
```diff
@@ -39,13 +39,17 @@ class DashboardControllerTest {
 
     @Test
     void dashboard() {
-        Dashboard dashboard = new Dashboard(9L, 99L, 999L);
+        Dashboard dashboard = new Dashboard(9L, 99L, 999L, 1L, 11L, 111L, 1111L);
```
[the remark this answers is not in the document]
Comment (ds-ext-sceronik) [reply tx-backend/src/test/java/org/eclipse/tractusx/traceability/assets/application/rest/DashboardControllerTest.java:42]: refactoring from java record to java class . then will be able to use builder pattern

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/qualitynotification/infrastructure/model/QualityNotificationStatusBaseEntity.java [hunk 7]
```diff
@@ -70,4 +73,10 @@ public static QualityNotificationStatusBaseEntity fromStringValue(String value)
         return MAPPINGS.get(value);
     }
 
+    public static QualityNotificationStatusBaseEntity from(QualityNotificationStatus status) {
```
[the remark this answers is not in the document]
Comment (ds-ext-sceronik) [reply tx-backend/src/main/java/org/eclipse/tractusx/traceability/qualitynotification/infrastructure/model/QualityNotificationStatusBaseEntity.java:76]: refactored all QualityNotification related entities to Notification :)

**Review by ds-lcapellino — APPROVED** [review]
LGTM

**Pull request merged**